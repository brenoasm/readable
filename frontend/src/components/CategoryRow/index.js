import React from 'react';
import styled from 'styled-components';
import { Link, withRouter } from 'react-router-dom';

import { colors } from '../../theme';

const StyledCategoryRow = styled.div`
  width: 100%;
  min-height: 100px;
  line-height: 100px;
  box-shadow: 0 1px 3px ${colors.primary.primaryThree};
  padding: 20px;
  text-align: center;
  vertical-align: middle;

  &:hover, &.active {
    border-left: 5px solid ${colors.primary.primaryThree};
  }

  a {
    font-size: 28px;
    text-decoration: none;
    color: ${colors.black};
  }
`;

const CategoryRow = ({ category, location }) => {
  const isActive = location.pathname === `/${category.path}`;

  return (
    <StyledCategoryRow className={isActive ? 'active' : ''}>
      <Link to={category.path}>
        {category.name}
      </Link>
    </StyledCategoryRow>
  );
};

export default withRouter(CategoryRow);
