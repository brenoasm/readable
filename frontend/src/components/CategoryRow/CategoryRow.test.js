/* eslint-disable */

import React from 'react';

import CategoryRow from './';

import CategoryMock from '../../testHelpers/mocks/categories';

describe('CategoryRow', () => {


  it('matches the snapshot', () => {
    const initialState = {
      category: CategoryMock.categories[0],
      activeRoute: '/redux'
    };

    const wrapper = shallow(<CategoryRow {...initialState} />);

    expect(wrapper).toMatchSnapshot();
  })

  it('should be the active route', () => {
    const initialState = {
      category: CategoryMock.categories[1],
      activeRoute: '/redux'
    };

    const wrapper = shallow(<CategoryRow {...initialState} />);

    expect(wrapper.hasClass('active')).toEqual(true);
  })

  it('should not be the active route', () => {
    const initialState = {
      category: CategoryMock.categories[1],
      activeRoute: '/react'
    };

    const wrapper = shallow(<CategoryRow {...initialState} />);

    expect(wrapper.hasClass('active')).toEqual(false);
  })
});
