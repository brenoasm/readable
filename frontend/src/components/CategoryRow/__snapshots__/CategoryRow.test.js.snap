// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryRow matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CategoryRow
    activeRoute="/redux"
    category={
        Object {
            "name": "react",
            "path": "react",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Link
        replace={false}
        to="react"
>
        react
</Link>,
      "className": "",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "react",
        "replace": false,
        "to": "react",
      },
      "ref": null,
      "rendered": "react",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Link
          replace={false}
          to="react"
>
          react
</Link>,
        "className": "",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "react",
          "replace": false,
          "to": "react",
        },
        "ref": null,
        "rendered": "react",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
